pkg/util/nodecache.go

```diff
@@ -102,9 +101,23 @@ func decodeObj(filepath, pattern string, file os.FileInfo, destObj interface{})
 
 }
 
+func (nc *NodeCache) Update(identifier string, data interface{}) error {
```

Comment:
Unless a lock is held during reading, modifying and writing the data, `Update` will always race - that's why it's missing here and it should stay that way. You _are_ holding a lock `volumeMountCacheMap` methods, but this locking is not explicit here in `NodeCache`. I'd suggest to remove this and create a separate update method in `volumeMountCacheMap` that would call `mc.NodeCacheStore.Delete()` and `mc.NodeCacheStore.Create()`.

Reply:
agree . i add new method updateNodeCache

https://github.com/huaizong/ceph-csi/blob/improve-remount-pv-path-when-exit-v2/pkg/cephfs/mountcache.go#L295-L305